Sources/StreamChatUI/ChatThread/ChatThreadVC.swift

```diff
@@ -445,12 +445,16 @@ open class ChatThreadVC: _ViewController,
         guard shouldRenderParentMessage else {
             return messageController.replies
         }
-        let messages = messageController.replies
+        var messages = messageController.replies
         let isFirstPage = messages.count < messageController.repliesPageSize
         let shouldAddRootMessageAtTheTop = isFirstPage || messageController.hasLoadedAllPreviousReplies
         if shouldAddRootMessageAtTheTop, let threadRootMessage = messageController.message {
             if messages.last?.id != threadRootMessage.id {
                 messages.append(threadRootMessage)
+            } else if messages.last?.id == threadRootMessage.id {
+                // When the last message is already the thread root message, update it.
+                let lastIndex = messages.count - 1
+                messages[lastIndex] = threadRootMessage
```

Comment:
This does not make much sense, we are checking if the `messages.last?.id` is the same as the `threadRootMessage.id`, but in fact, `messages.last?.id` in this case is the same as doing `messageController.replies.last?.id`. The replies should never include the parent message according to `MessageDTO.threadRepliesPredicate()`, but somehow, with BG Mapping disabled, the thread parent message is included in the replies when `didUpdateMessage` delegate is reported.

Reply:
weird 🤔 Anyway, no harm in adding this

Reply:
This seem to happen because we are inserting the parent message to messageController.replies here on this line
```if messages.last?.id != threadRootMessage.id {
    messages.append(threadRootMessage)
}```
messages here is `LazyCachedMapCollection<ChatMessage>`  and turns out that `LazyCachedMapCollection` has an append method which updates cached values.

```public func append(_ element: Element) {
  cache.values.append(element)
}```
`LazyCachedMapCollection`  in turn uses that cache property to return what it represents,  in this case: what FRC returned + the value we added ourselves.

Reply:
OMG, did not expect `cache` was a reference type... 😭 Will close this ticket, since the solution does not make sense anymore